### ISSUE TITLE
ALB-089: GPU-accelerated inference for eval (CUDA forward pass)

### ISSUE THREAD

**Issue by noahgift** [paiml/albor#69]
## Gap

Current eval uses realizar's CPU inference (AprTransformer::forward_with_cache). For a 350M model:
- CPU: ~18s per problem (256 tokens), ~50 min for 164 HumanEval problems
- With multi-sample (200×): ~164 hours — impractical

entrenar already has a full GPU forward pass (CudaTransformerTrainer). We need to expose GPU inference for eval.

## Acceptance Criteria

- `apr eval --task humaneval --data humaneval.jsonl --device cuda`
- Load SafeTensors weights to GPU via entrenar's CudaTransformerTrainer
- Run forward pass on GPU, copy logits to CPU for sampling
- Expected speedup: 10-20x over CPU (based on training throughput: 3.5K tok/s GPU vs ~200 tok/s CPU)
- HumanEval 164 problems in <5 min (vs 50 min CPU)
- Multi-sample 200× in <16 hours (vs 164 hours CPU)

## Architecture

Option A: Add `generate()` method to CudaTransformerTrainer
Option B: Create CudaInferenceEngine that loads weights without optimizer state

Option B preferred — inference needs only weights (1.4 GB), not optimizer state (4.2 GB).

## Dependencies

- ALB-088 (multi-sample pass@k) — GPU inference makes multi-sample practical
